frontend/pkg/frontend/frontend.go

```diff
@@ -211,7 +211,39 @@ func (f *Frontend) ArmResourceListBySubscription(writer http.ResponseWriter, req
 
 	f.logger.Info(fmt.Sprintf("%s: ArmResourceListBySubscription", versionedInterface))
 
-	writer.WriteHeader(http.StatusOK)
+	subscriptionId := request.PathValue(PathSegmentSubscriptionID)
+
+	query := fmt.Sprintf("azure.subscription_id='%s'", subscriptionId)
+	clustersListResponse, err := f.conn.ClustersMgmt().V1().Clusters().List().Search(query).Send()
+	if err != nil {
+		f.logger.Error(err.Error())
+		arm.WriteInternalServerError(writer)
+		return
+	}
+
+	var hcpCluster *api.HCPOpenShiftCluster
+	clusters := clustersListResponse.Items().Slice()
+	for _, cluster := range clusters {
+		hcpCluster, err = f.ConvertCStoHCPOpenShiftClusterbySubscription(cluster)
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+
+		versionedResource := versionedInterface.NewHCPOpenShiftCluster(hcpCluster)
+		resp, err := json.MarshalIndent(versionedResource, "", "    ")
+		if err != nil {
+			f.logger.Error(err.Error())
+			arm.WriteInternalServerError(writer)
+			return
+		}
+		_, err = writer.Write(resp)
+		if err != nil {
+			f.logger.Error(err.Error())
+		}
+	}
+
```

Comment:
```suggestion
  writer.WriteHeader(http.StatusOK)
```

Reply:
Kept getting `http: superfluous response.WriteHeader call` with that line